controllers/ingressnodefirewall_controller.go

```diff
@@ -18,22 +18,32 @@ package controllers
 
 import (
 	"context"
+
 	"github.com/go-logr/logr"
-	"github.com/pkg/errors"
-	nodefwloader "ingress-node-firewall/pkg/ebpf"
 
 	ingressnodefwv1alpha1 "ingress-node-firewall/api/v1alpha1"
-	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
 // IngressNodeFirewallReconciler reconciles a IngressNodeFirewall object
 type IngressNodeFirewallReconciler struct {
 	client.Client
-	Scheme *runtime.Scheme
-	Log    logr.Logger
+	Scheme    *runtime.Scheme
+	Log       logr.Logger
+	Namespace string
```

Comment:
this is cluster scoped object why we passing ns here ?

Reply:
It's for listing the NodeState objects which are namespaced. By the way:
https://github.com/openshift/ingress-node-firewall/issues/54

Reply:
yes fw is cluster scope , I wanted config to be also cluster scope but I think Own didn' work and had to switch to ns